frontend/app/src/main/java/com/example/speechbuddy/viewmodel/SignupViewModel.kt

```diff
@@ -60,17 +63,26 @@ class SignupViewModel @Inject internal constructor(
     }
 
     private fun validateNickname() {
-        if (nicknameInput.isNotEmpty() && nicknameInput.length <= Constants.MAXIMUM_NICKNAME_LENGTH) {
+        if (isValidNickname(nicknameInput)) {
             _uiState.update { currentSate ->
                 currentSate.copy(
                     isValidNickname = true,
                     error = null
                 )
             }
+        } else {
+            _uiState.update { currentState ->
+                currentState.copy(
+                    isValidNickname = false,
+                    error = SignupError(
+                        type = SignupErrorType.NICKNAME,
+                        messageId = R.string.nickname_too_long
+                    )
+                )
+            }
```

Comment:
```suggestion
        } else if (nicknameInput.isEmpty()) {
            _uiState.update { currentState ->
                currentState.copy(
                    isValidNickname = false,
                    error = SignupError(
                        type = SignupErrorType.NICKNAME,
                        messageId = R.string.no_nickname
                    )
                )
            }
        } else {
            _uiState.update { currentState ->
                currentState.copy(
                    isValidNickname = false,
                    error = SignupError(
                        type = SignupErrorType.NICKNAME,
                        messageId = R.string.nickname_too_long
                    )
                )
            }
```
닉네임이 입력되지 않았을 때, 15자 초과일 때 두 상황의 에러메시지 분리했습니다!